### PR TITLE
synchronize napoleon rst and python docs

### PR DESCRIPTION
**Pull request by svenevs**
* napoleon_include_special_with_doc defaults have changed
* add documentation for napoleon_custom_sections

-------------------

Docs are slightly out of sync, I copy-pasted and reformatted as things were.  Only deviation was

1. Remove `\"` in one place (needed in python docs, not rst).
2. Seems that somebody re-flowed `napoleon_use_keyword` docs when converting pydocs to rst.  It only affects one paragraph, so I "undid" that to avoid diffs in future.

### CODE REVIEW

**Comment by svenevs** [comment]
Ok so the first three commits can be cherry-picked at leisure to fix napoleon and autodoc docs.

The fourth commit `[proposal]` is the implementation I'm proposing we adopt, rationale explained in https://github.com/sphinx-doc/sphinx/pull/7018#issuecomment-578581257 .  The Type Annotations docs for the NumPy are included, since those docs change depending on what we decide to do about the rtype.

Will wait for your response, you also have write access to my branch so please feel free to change as desired and then squash things :)

File: doc/usage/extensions/napoleon.rst [hunk 1]
```diff
@@ -496,3 +535,36 @@ sure that "sphinx.ext.napoleon" is enabled in `conf.py`::
...
+
+        Returns
+        -------
+      + A description of the return.
+      - bool
+      -     A description of the return.
+
+   As python function signatures may only have one return type annotation,
+   setting this value to True implies only one return type is being
+   documented.
```
Comment (svenevs) [inline doc/usage/extensions/napoleon.rst:567]: Is it necessary to clarify with an example of returning with `Union[int, str]` or is this clear enough?  It's already kind of long-winded...

File: doc/usage/extensions/autodoc.rst [hunk 2]
```diff
@@ -437,10 +437,15 @@ There are also config values that you can set:
    following values:
 
    * ``'signature'`` -- Show typehints as its signature (default)
+   * ``'description'`` -- Inject ``:type:`` and ``:rtype:`` annotations into the
+     documentation string.
```
Comment (tk0miya) [inline doc/usage/extensions/autodoc.rst:441]: This is not adopted to formal option. It is only available when `sphinx.ext.autodoc.typehints` loaded. So I'll add this in 3.0 release finally.

File: doc/usage/extensions/napoleon.rst [hunk 3]
```diff
@@ -478,3 +521,50 @@ sure that "sphinx.ext.napoleon" is enabled in `conf.py`::
    **If False**::
 
        :returns: *bool* -- True if successful, False otherwise
+
+.. confval:: napoleon_custom_sections
+
+   Add a list of custom sections to include, expanding the list of parsed sections.
+   *Defaults to None.*
```
Comment (tk0miya) [inline doc/usage/extensions/napoleon.rst:528]: You don't need to emphasis this sentence. Please make it plain.

File: doc/usage/extensions/napoleon.rst [hunk 4]
```diff
@@ -478,3 +521,50 @@ sure that "sphinx.ext.napoleon" is enabled in `conf.py`::
...
+
+   The entries can either be strings or tuples, depending on the intention:
+     * To create a custom "generic" section, just pass a string.
+     * To create an alias for an existing section, pass a tuple containing the
+       alias name and the original, in that order.
```
Comment (tk0miya) [inline doc/usage/extensions/napoleon.rst:533]: This list should not be indented. And blank line is needed before the list.

File: doc/usage/extensions/napoleon.rst [hunk 5]
```diff
@@ -478,3 +521,50 @@ sure that "sphinx.ext.napoleon" is enabled in `conf.py`::
    **If False**::
 
        :returns: *bool* -- True if successful, False otherwise
+
+.. confval:: napoleon_custom_sections
```
Comment (tk0miya) [inline doc/usage/extensions/napoleon.rst:525]: I think this is not related with this PR. Could you make another one please?

File: doc/usage/extensions/napoleon.rst (hunk 1 above)
Comment (tk0miya) [reply doc/usage/extensions/napoleon.rst:567]: Indeed. It would be better to do that, but it is a bit difficult...

File: doc/usage/extensions/napoleon.rst [hunk 6]
```diff
@@ -478,3 +521,50 @@ sure that "sphinx.ext.napoleon" is enabled in `conf.py`::
...
+
+   If an entry is just a string, it is interpreted as a header for a generic
+   section. If the entry is a tuple/list/indexed container, the first entry
+   is the name of the section, the second is the section key to emulate.
+
+.. confval:: napoleon_numpy_returns_no_rtype
+
+   *Defaults to False.*
+   Traditional `NumPy style`_ requires the return type and a description
+   provided indented underneath::
```
Comment (tk0miya) [inline doc/usage/extensions/napoleon.rst:543]: Is it needed to contact numpy team before adding this? I think this format is theirs. We should not customize it.
https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard

**Comment by tk0miya** [comment]
Thank you for proposal. It seems this is not a blocker to merge typehints extension. So I'll merge the original PR sooner. After that, it would be better to discuss in another PR on sphinx repo, not mine.
Thanks!

**Comment by svenevs** [comment]
Ok I will address the comments in separate PRs post-merge.  I will also reach out to NumPy people to solicit input from them.
